docs/functions-dev.md

```diff
@@ -42,3 +42,66 @@ To do so set the `--inspect` Node.js option when starting the functions server:
 - On Mac/Linux run `NODE_OPTIONS=--inspect netlify functions:serve`
 
 Then attach any Node.js debugger to the CLI process to debug your functions.
+
+
+## VS Code Debugging
+
+Put the below configuration in your repo's `.vscode/launch.json`.
+
+Then in `VS Code`:
+1. Open `Run and Debug` Sidebar. Make sure "Caught Exceptions" is deactivated.
+2. In the dropdown, select `Fullstack Debugging`
+3. Run the debugger. A chrome window should open. Wait until content is loaded (may take a few seconds for server to be ready).
+
+```jsonc
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  // ---- FULLSTACK NETLIFY DEV DEBUGGING ----
+  // 1. Make sure "Caught Exceptions" is deactivated in your VS Code debugger
+  // 2. Run 'Fullstack Debugging' from the debug interface
+  // 3. A chrome browser window will open, however it may take a few seconds
+  // before the server is ready and content is displayed.
+  "compounds": [
+    {
+      "name": "Fullstack Debugging",
+      "configurations": ["netlify dev", "client: chrome"]
+    }
+  ],
+  "configurations": [
+    {
+      "name": "client: chrome",
+      "type": "chrome",
+      "request": "launch",
+      "url": "http://localhost:8888",
+      "webRoot": "${workspaceFolder}",
+      "sourceMapPathOverrides": {
+        "webpack:///*": "${workspaceRoot}/*"
+      }
+    },
+    {
+      "name": "netlify dev",
+      "type": "node",
+      "request": "launch",
+      "skipFiles": ["<node_internals>/**"],
+      "program": "${workspaceFolder}/node_modules/.bin/netlify",
+      "args": ["dev", "--inspect"],
+      "resolveSourceMapLocations": [
```

Comment:
I don't think this is required as a default configuration.

I would rather have another section that explains about source maps and the purpose of this setting, as this is a more advanced use case.

Reply:
As I understand, netlify dev is used with js/ts/node. I would think then that a) using node modules would be pretty default for this use case b) as is the need to exclude them from source maps.

For me at least, without this source-map config, I get lots of errors in the debug console on startup.  I remember also reading that without proper sourcemap config breakpoints might not work (I didn't test that).

However, also happy to spawn this out into it's own section if you prefer. :)

What concern me a more however is, that I noticed that errors are not forwarded to `debug console` or vs code.

Reply:
> As I understand, netlify dev is used with js/ts/node

I'm not sure I understand this statement. `netlify dev` runs as a regular Node.js executable and doesn't require any source maps.

> For me at least, without this source-map config, I get lots of errors in the debug console on startup. I remember also reading that without proper sourcemap config breakpoints might not work (I didn't test that).
> 
> However, also happy to spawn this out into it's own section if you prefer. :)

Could this be related to frontend debugging? Have you seen this for functions too? If so, can you share an example repo? By default, breakpoints in JavaScript functions should work.

Reply:
I think VS code debugger may need access to the sourcemaps.

The error messages are showing up in the netlify dev debug console, not the client console, so definitely backend related.

You can reproduce it by using my MRE https://github.com/MentalGear/netlify-dev_debug_vs-code/issues and comment out in `launch.json`:
- `resolveSourceMapLocations `
- "console": "integratedTerminal" (this I added later as vs-code `debug console` doesn't forward all the std output as integratedTerminal does)

```jsonc
      // "console": "integratedTerminal" // avoids logs & errors from being suppressed
      // "resolveSourceMapLocations": [
      //   // NOTE: your linter may think this property shouldn't be here
      //   // yet it works for finding the source maps
      //   "${workspaceFolder}/**",
      //   "!**/node_modules/**"
      // ]
```